api/v1alpha2/terraform_types.go

```diff
@@ -826,6 +845,30 @@ func TerraformStateLocked(terraform Terraform, lockID, message string) Terraform
 	return terraform
 }
 
+// TerraformReachedLimit will set a new condition on the Terraform resource
+// indicating that the resource has reached its retry limit.
+func TerraformReachedLimit(terraform Terraform) Terraform {
+	newCondition := metav1.Condition{
+		Type:    ConditionRetryLimitReached,
```

Comment:
First I used Stalled and then somewhere I found an example to use `Status: ConditionFalse` for retry resets, and then I found a different one that used custom type, but remove after reset and... well I ended up with a mix of them it seems.

And it was easier to just check that specific Status in the Conditions slice, rather than checking if it's Stalled and has given reason (not it's not really, just an extra `&& status.Reason == infra.XXX`, but feels like it's more complex).

▶️ I'll change it to Stalled and move the reason into a constant.